src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3997,7 +4022,7 @@ private void AddDisposingFinally(IOperation resource, bool requiresRuntimeConver
 
                 if (method != null)
                 {
-                    var args = disposeMethod is object ? VisitArguments(disposeArguments) : ImmutableArray<IArgumentOperation>.Empty;
+                    var args = VisitArray(disposeArguments, UnwrapArgument, RewriteArgumentFromArray);
```

Comment:
> var args = VisitArray(disposeArguments, UnwrapArgument, RewriteArgumentFromArray); [](http://example.com/codeflow?start=19&length=83)

It is not obvious why we need to make changes here, especially removal of the conditional logic looks suspicious

Reply:
I think that in the long term it would be better to keep calling VisitArguments with appropriate value for the second parameter.